java/src/eda/sorting/InsertionSort.java

```diff
@@ -0,0 +1,51 @@
+package eda.sorting;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+/**
+ * Created by JoseManuel on 19/04/2016.
+ */
+public class InsertionSort<T extends Comparable<T>> implements Sorting<T> {
```

Comment:
If InsertionSort methods are statics, You don't need create generic class, only in the methods declaration.